### ISSUE TITLE
Off-beat samples in song editor snap to grid when moving them with a selection

### ISSUE THREAD

**Issue by MightyPork** [LMMS/lmms#2300]
Really annoying when trying to make something like trance, with reverse cymbals preceding start of drums section etc. They often have "peak" somewhere in the middle, so I have to adjust their position with CTRL-drag.

But then I decide to move the whole part a bit, and the fine adjustments are lost :(

Here's a screen capture of the bug happening: [video on youtube](https://youtu.be/FyHTOKXgxFQ)

I'd help, if it wasn't in C++ :(

**Comment by tresf** [comment]
Isn't this what reverse-delay is for?

Duplicate of #87 I believe.

-Tres

**Comment by MightyPork** [comment]
I don't think so, it's more of a usability problem (position gets rounded when moving).

Reverse delay would sort of work - but if I have more than one sample in the track, they may need different offsets = no go. I can't even adjust individual samples' volume (which is a big WTF for me, but can be worked around by putting them in separate tracks — which in turn causes UI lag, but not too bad).

**Comment by tresf** [comment]
@MightyPork can you agree that it is a duplicate of #87?

**Comment by MightyPork** [comment]
No, it is not a duplicate.

**Comment by tresf** [comment]
Confirmed, thanks for clarification.  All off-beat patterns in song editor in `stable-1.1` branch have no way to move without snapping.

I believe this is actually a bug with our default shortcuts and we'll need to confirm this in the `master` branch so make sure it hasn't already been fixed.

**Comment by tresf** [comment]
FYI, a workaround is to add new bars and remove bars using shortcut keys.  Not ideal, but can help.  I believe the keys are CTRL + SHIFT + INSERT, CTRL + SHIFT + DELETE.

Also, this is a timing thing... if you click but don't drag left or right, then press the modifier key,  this will allow you to move the segment and preserve fine-tuned-positions.  Of course, workflow almost never honors this because you move left or right first.

Secondly, many of us don't write C++, so if you're ok with shuffling a few variables, you could help this bug without having to take a C++ class. :+1: 

Last, thanks for taking the time to upload a video, much obliged.

**Comment by tresf** [comment]
Also, using an AudioFile processor for now will allow you to align those samples (manual and no visual indicator) but can let the project survive this bug until we have the behavior fixed.  Getting the position right is brutal, but at least it survives the moves.  I can take a screenshot if this description doesn't make sense.
